src/Rings/mpoly-ideals.jl

```diff
@@ -352,11 +352,12 @@ Ideal generated by
 ```
 """
 function saturation(I::MPolyIdeal{T}, J::MPolyIdeal{T} = ideal(base_ring(I), gens(base_ring(I))); iteration::Bool=false) where T
-  # `remove` is faster than `Singular.saturation` when saturating with
+  # `ppio` is faster than `Singular.saturation` when saturating with
   # respect to principal ideals
-  if ngens(I) == 1 && ngens(J) == 1
-    is_unit(J[1]) && return I
-    return ideal(remove(I[1], J[1])[2])
+  if is_known(is_principal, I) && is_principal(I) && is_known(is_principal, J) && is_principal(J)
+    is_unit(principal_generator(I)) && return I
+    is_gen(principal_generator(I)) && ideal(remove(principal_generator(I), principal_generator(J))[2]) # catches a special case in strict transforms of toric blowups 
```

Comment:
```suggestion
    is_gen(principal_generator(I)) && return ideal(remove(principal_generator(I), principal_generator(J))[2]) # catches a special case in strict transforms of toric blowups 
```
I think you mean this